[PATCH] setserial: fix build with 2.6.33+ kernel headers

linux/hayesesp.h got removed in 2.6.33, but some archs (like x86) still
define the TIOCGHAYESESP ioctl, breaking the build.

It's very unlikely to be of interest anymore, so just undef it.

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
[rebased on top of Debian's setserial_2.17-45.3.diff]
Upstream: N/A only for compatibility with Linux kernels [2.6.33;2.6.36[
Signed-off-by: Quentin Schulz <quentin.schulz@cherry.de>
---
 setserial.c |    5 +++++
 1 file changed, 5 insertions(+)

Index: setserial-2.17/setserial.c
===================================================================
--- setserial-2.17.orig/setserial.c
+++ setserial-2.17/setserial.c
@@ -39,6 +39,11 @@
 #include "safety.h"
 #endif
 
+/* linux/hayesesp.h got removed in 2.6.33 but some archs (like x86)
+   still define TIOCGHAYESESP. It's very unlikely to be of interest
+   anymore, so just undef it */
+#undef TIOCGHAYESESP
+
 #include "version.h"
 
 static char version_str[] = "setserial version " SETSERIAL_VERSION ", "
